### PR TITLE
log detail errors from remote

### PR DESCRIPTION
**Pull request by jagedn**
in some situations we need more details about the error returned by the server so log them

### CODE REVIEW

File: plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy [hunk 1]
```diff
@@ -300,12 +316,21 @@ class NomadService implements Closeable{
 
     protected void purgeJob(String jobId, boolean purge){
         log.debug "[NOMAD] purgeJob with jobId=${jobId}"
-        jobsApi.deleteJob(jobId,config.jobOpts().region, config.jobOpts().namespace,null,null,purge, true)
+        try {
+            jobsApi.deleteJob(jobId, config.jobOpts().region, config.jobOpts().namespace, null, null, purge, true)
+        }catch(Exception e){
+            log.error("[NOMAD] Failed to delete job ${jobId} -- Cause: ${e.message ?: e}", e)
```
Comment (abhi18av) [inline plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy:322]: @jagedn , do you think that within an exception block, we should use only `log.debug` or this should be decided as per the context?

I think `error` makes sense for session terminating steps and not for the retries - what do you think?

**Review by abhi18av — APPROVED** [review]
Looks good to merge - Thanks @jagedn!

I have added a question regarding the overall best-practice for `log.debug` and `log.error` .

File: plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy (hunk 1 above)
Comment (jagedn) [reply plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy:322]: good question

yes, you're right, probably we can use only debug level in retries and if the user want to have more details can change the level

Comment (abhi18av) [reply plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy]: Thanks Jorge, I've refactored accordingly 👍

**Pull request merged**